### ISSUE TITLE
third party activities crash

### ISSUE THREAD

**Issue by rbyakov** [pandulapeter/beagle#82]
Hey, can you add opposite of `Behavior.excludedPackageNames`. It would be nice especially for single activity applications. Kind of `Behavior(resolvedActivities = listOf("com.my.package.MainActivity"))`. Because it's hard for supporting list of strings of third-party libs

**Comment by pandulapeter** [comment]
Hi!

Thanks for raising this point. I can add something similar in the next version (while still supporting the old implementation). The plan could be to keep `excludedPackageNames` for backwards compatibility and add this new `resolvedActivities`  list that you proposed. Both would be `null` by default, so the following situations could appear:

1) `excludedPackageNames == null && resolvedActivities == null` -> The library tries to inject the debug menu in every Activity
2) `excludedPackageNames != null && resolvedActivities == null`  -> The library tries to inject the debug menu in every Activity except for the ones specified in `excludedPackageNames`
3) `excludedPackageNames == null && resolvedActivities != null`  -> The library tries to inject the debug menu only in the activities specified in `resolvedActivities`
4) `excludedPackageNames != null && resolvedActivities != null` -> This would be an invalid state I guess, maybe we should throw an exception

What do you think, is this what you had in mind?

**Comment by pandulapeter** [comment]
I introduced a new way to approach this whole thing in [release 2.6.0](https://github.com/pandulapeter/beagle/releases/tag/2.6.0).

Unfortunately it's a breaking change to the API, but it's cleaner than what was described above.

The `Behavior.excludedPackageNames` list has been replaced with a lambda property: `Behavior.shouldAddDebugMenu`. This function will be invoked with a `FragmentActivity` argument every time when Beagle tries to inject itself into a new Activity. If the implementation returns `false`, the debug menu will not be present on that screen. Nevertheless, the library internally checks against a list of known, unsupported third-party Activities before invoking the consumer implementation. The parameter is optional, the default implementation returns `true` for any input argument (which should be enough for most apps).
